src/target/riscv/riscv.c

```diff
@@ -1946,14 +1956,15 @@ int riscv_set_current_hartid(struct target *target, int hartid)
 	if (!target_was_examined(target))
 		return ERROR_OK;
 
+#if 0
```

Comment:
Seems a bit heavy-handed, but given that hopefully we'll switch away from `-rtos riscv` it might not be worth the effort to fix properly.

If we are removing this code, let's remove it instead of using `#if 0`.

Reply:
I think this might be masking another issue, but I'm not sure.  If it's not then removing it seems fine.

Reply:
Remove the entire #if 0 block.